src/catchup/CatchupMinimalWork.cpp

```diff
@@ -104,42 +99,23 @@ CatchupMinimalWork::onSuccess()
         return WORK_PENDING;
     }
 
-    // Phase 4: download and verify the buckets themselves.
-    if (!mDownloadBucketsWork)
+    // Phase 4: download, verify and apply buckets themselves.
+    if (!mDownloadAndApplyBucketsWork)
     {
         CLOG(INFO, "History")
-            << "Catchup MINIMAL downloading and verifying buckets";
+            << "Catchup MINIMAL downloading, verifying and applying buckets";
         std::vector<std::string> buckets =
             mGetHistoryArchiveStateWork->getRemoteState().differingBuckets(
                 mLocalState);
-        mDownloadBucketsWork = addWork<Work>("download and verify buckets");
-        for (auto const& hash : buckets)
-        {
-            FileTransferInfo ft(*mDownloadDir, HISTORY_FILE_TYPE_BUCKET, hash);
-            // Each bucket gets its own work-chain of download->gunzip->verify
-
-            auto verify = mDownloadBucketsWork->addWork<VerifyBucketWork>(
-                mBuckets, ft.localPath_nogz(), hexToBin256(hash));
-            verify->addWork<GetAndUnzipRemoteFileWork>(ft);
-        }
+        mDownloadAndApplyBucketsWork = addWork<DownloadAndApplyBucketsWork>(
```

Comment:
not a fan of this refactoring if (and it is) is only used in one place. It just adds more work types, and we have already too many.